### PR TITLE
Egui 0.3 update

### PR DESCRIPTION
**Pull request by Gonkalbell**
This updates `egui` to 0.3, and adds support for user textures.

Here are the pull requests for updating the 2 dependencies
https://github.com/hasenbanck/egui_wgpu_backend/pull/2
https://github.com/hasenbanck/egui_winit_platform/pull/1

### CODE REVIEW

**Comment by Gonkalbell** [comment]
By the way, have you considered moving the `wgpu` bindings and `winit` bindings in the same repository, but kept them as separate crates in the same workspace? I think that would make updating these crates easier to manage, while you still get to keep them as separate crates on crates.io

File: Cargo.toml [hunk 1]
```diff
@@ -6,18 +6,18 @@ edition = "2018"
 
 [dependencies]
 chrono = "0.4"
-futures-lite = "1.10"
-egui = "0.2"
-egui_wgpu_backend = "0.1"
-egui_winit_platform = "0.1"
+futures-lite = "1.11"
+egui = "0.3"
+egui_wgpu_backend = { version = "0.2", git = "https://github.com/hasenbanck/egui_wgpu_backend" }
```
Comment (hasenbanck) [inline Cargo.toml:11]: Remove the override here and localy use the [patch.crates-io] or the crate specific patch please.

**Comment by hasenbanck** [comment]
@Gonkalbell Since the crates don't share any code and are seperate from each other (WGPU and WINIT) it doesn't make much sense to force them into one repository.

Could you please apply "cargo fmt" on all of your PRs?

File: Cargo.toml (hunk 1 above)
Comment (Gonkalbell) [reply Cargo.toml]: Fixed. I was confused before because I didn't understand how the `patch` property worked, but it should be correct now

**Comment by Gonkalbell** [comment]
I ran `cargo fmt` on everything. I forgot that I had `use_small_heuristics = "Max"` in my personal rustfmt.toml, so that's why there were so many formatting changes before. I set it back to the default, so now there should be less formatting noise in the other PRs

**Pull request merged**